src/helper/HelperApp.cpp

```diff
@@ -167,19 +167,6 @@ namespace SDDM {
         m_user = m_backend->userName();
         QProcessEnvironment env = authenticated(m_user);
 
-        if (env.value(QStringLiteral("XDG_SESSION_CLASS")) == QLatin1String("greeter")) {
-            // Qt internally may load the xdg portal system early on, prevent this, we do not have a functional session running.
-            env.insert(QStringLiteral("QT_NO_XDG_DESKTOP_PORTAL"), QStringLiteral("1"));
-            for (const auto &entry : mainConfig.GreeterEnvironment.get()) {
-                const int index = entry.indexOf(QLatin1Char('='));
-                if (index < 0) {
-                    qWarning() << "Malformed environment variable" << entry;
-                    continue;
-                }
-                env.insert(entry.left(index), entry.mid(index + 1));
-            }
-        }
-
         if (!m_session->path().isEmpty()) {
             env.insert(m_session->processEnvironment());
             m_session->setProcessEnvironment(env);
```

Comment:
What do you think about just moving those assignments here, after the `env.insert(m_session->processEnvironment());` call? That should have the same effect but is IMO a slightly more fitting place

Reply:
Not entirely the same, `getpwnam` and `login_getpwclass` will have the priority over GreeterEnvironment but will still help with the exact case of QT_PLUGIN_PATH in PAM though.

Reply:
Yeah, I don't really know what `setclassenvironment` does on FreeBSD. Maybe it should override GreeterEnvironment, maybe it shouldn't...

Reply:
I'd say it has sense for GreeterEnvironment to override anything else

Reply:
Ok.